test/simulator/test/org/apache/cassandra/simulator/paxos/HistoryValidatorTest.java

```diff
@@ -356,79 +383,146 @@ private static Event writeOnly(int pk)
         return new Event(EnumSet.of(Event.Type.WRITE), pk, null);
     }
 
-    private void fromLog(String log)
+    private interface Operation
     {
-        IntSet pks = new IntHashSet();
-        class Read
+        int pk();
+        void check(HistoryValidator.Checker check);
+        void toString(StringBuilder sb);
+    }
+
+    private static class Read implements Operation
+    {
+        final int pk, id, count;
+        final int[] seq;
```

Comment:
more verbose and have no impact.  Since all access is in the same file java will allow all classes to reach in and touch...

now, when it comes to our style, when its not directly needed we should remove; so in this case since `private` doesn't matter our style says it shouldn't be there.